### PR TITLE
Add doc about content verification

### PR DESCRIPTION
**Pull request by ktock**
Related: #114

### CODE REVIEW

File: docs/verification.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# Content Verification in eStargz
+
+The goal of the verifiability in eStargz is to make all files' metadata and contents in an eStargz layer verifiable.
```
Comment (AkihiroSuda) [inline docs/verification.md:3]: Definition of "verify"?

File: docs/verification.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
+# Content Verification in eStargz
```
Comment (AkihiroSuda) [inline docs/verification.md:1]: link this file from `stargz-estargz.md`

File: docs/verification.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+`ctr-remote images optimize` command in this project creates the verifiable eStargz image by default.
+
+## Content verification in Stargz Snapshotter
+
+Stargz Snapshotter verifies eStargz layers leveraging the above extensions.
+However, as mentioned in the above, the verification of other image component including the manifests is out-of-scope of the snapshotter.
+So when this snapshotter mounts an eStargz layer, the manifest that references this layer must be verified in the containerd in advance and the TOC's digest written in the manifest (as an annotation `containerd.io/snapshot/stargz/toc.digest`) must be passed down to this snapshotter.
+
+If the manifest doesn't contain `containerd.io/snapshot/stargz/toc.digest` annotation, verification can't be done for that layer so stargz snapshotter reports an error and doesn't mount it.
+You can bypath this check only if both of the following conditions meet.
```
Comment (AkihiroSuda) [inline docs/verification.md:49]: bypass

File: docs/verification.md (hunk 1 above)
Comment (ktock) [reply docs/verification.md:3]: Through the series of commits related #114, what I aimed to was enabling:

"ensuring the downloaded metadata and contents of all files are the expected ones, based on the calculated OCI digests"
And I called it "verification" so far. 

So the first line should be:
```
The goal of the content verification in eStargz is to ensure the downloaded metadata and contents of all files are the expected ones, based on the calculated OCI digests.
```

Please tell me if I'm missing something or there is something to be added/fixed in the definition and/or the implementation.

Comment (ktock) [reply docs/verification.md:3]: And thanks for the comment! :pray:

**Pull request merged**